### PR TITLE
feat: improvement of sign up fragment

### PR DESCRIPTION
**Pull request by pranavpandey1998official**
Feat: Enhancing the signup experience  

1. Setting the inputType property of lastName editTest and FirstName editText to textCapWords so the  first letter of first name and last name is capitalized
2.Setting the imeOption of lastName to actionDone 
3.Implementing the setOnEditorActionListener of lastName editText in SignUpFragment to handle submitting the form and also auto-hiding the keyboard 
 
feat: #834

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/auth/SignUpFragment.kt [hunk 1]
```diff
@@ -19,10 +22,7 @@ import kotlinx.android.synthetic.main.fragment_signup.passwordSignUp
 import kotlinx.android.synthetic.main.fragment_signup.signUpButton
 import kotlinx.android.synthetic.main.fragment_signup.textInputLayoutPassword
 import kotlinx.android.synthetic.main.fragment_signup.usernameSignUp
-import kotlinx.android.synthetic.main.fragment_signup.view.passwordSignUp
-import kotlinx.android.synthetic.main.fragment_signup.view.progressBarSignUp
-import kotlinx.android.synthetic.main.fragment_signup.view.signUpButton
-import kotlinx.android.synthetic.main.fragment_signup.view.signupCoordinatorLayout
+import kotlinx.android.synthetic.main.fragment_signup.view.*
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/auth/SignUpFragment.kt]: Don't use wildcard imports. @iamareebjamal Why build is successful with it?

File: app/src/main/java/org/fossasia/openevent/general/auth/SignUpFragment.kt [hunk 2]
```diff
@@ -45,6 +45,18 @@ class SignUpFragment : Fragment() {
         setHasOptionsMenu(true)
         val signUp = SignUp()
 
+        rootView.lastNameText.setOnEditorActionListener { v, actionId, event ->
+            return@setOnEditorActionListener when (actionId) {
+                EditorInfo.IME_ACTION_DONE -> {
+                    rootView.signUpButton.performClick()
+                    val imm = v.context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+                    imm.hideSoftInputFromWindow(v.windowToken, 0)
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/auth/SignUpFragment.kt]: This method is already written in utils. You can call it from there.

File: app/src/main/java/org/fossasia/openevent/general/auth/SignUpFragment.kt [hunk 3]
```diff
@@ -101,9 +114,11 @@ class SignUpFragment : Fragment() {
                 }
             }
 
-            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) { /*Implement here*/ }
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) { /*Implement here*/
+            }
 
-            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) { /*Implement here*/ }
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) { /*Implement here*/
+            }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/SignUpFragment.kt]: Unnecessary changes
